auth0/v3/authentication/token_verifier.py

```diff
@@ -340,3 +339,135 @@ def _verify_payload(self, payload, nonce=None, max_age=None, organization=None):
+        verify the signature.
+        issuer (str): The expected issuer claim value.
+        audience (str): The expected audience claim value.
+        leeway (int, optional): The clock skew to accept when verifying date
+        related claims in seconds. Defaults to 60 seconds.
+    """
+
+    def __init__(self, signature_verifier, issuer, audience, leeway=0):
+        instance_check = isinstance(signature_verifier, SignatureVerifier)
+        if not signature_verifier or not instance_check:
+            raise TypeError("signature_verifier must be an instance of "
+                            "SignatureVerifier.")
+
+        self.iss = issuer
+        self.aud = audience
+        self.leeway = leeway
+        self._sv = signature_verifier
+        self._clock = None  # visible for testing
+
+    def verify(self, token, scopes=None, permissions=None):
+        """Attempts to verify the given JWT access token, following the steps
+        defined in the RFC 7519, for more details see:
+        https://datatracker.ietf.org/doc/html/rfc7519#section-7.2
+        Auth0 only uses JWT access tokens.
+
+        Args:
+            token (str): The JWT access token to verify.
+            scopes (str, optional): A string of scopes to verify.
+            permissions (list(str), optional): A list of permissions to verify.
+
+        Raises:
+            TokenValidationError: when the token cannot be decoded, the token signing algorithm is not the expected one,
+            the token signature is invalid or the token has a claim missing or with unexpected value.
+
+        Returns:
+            If all checks pass, the token payload is returned
+        """
+
+        # Verify token presence
+        if not token or not isinstance(token, str):
+            raise TokenValidationError("Access token is required but missing.")
+
+        # Verify algorithm and signature
+        payload = self._sv.verify_signature(token)
+
+        # Verify claims
+        self._verify_payload(payload, scopes, permissions)
+
+        return payload
+
+    def _verify_payload(self, payload, scopes=None, permissions=None):
+        try:
+            # on Python 2.7, 'str' keys as parsed as 'unicode'
+            # But 'unicode' was removed on Python 3.7
+            # noinspection PyUnresolvedReferences
+            ustr = unicode
+        except NameError:
+            ustr = str
+
+        # Issuer
+        if 'iss' not in payload or not isinstance(payload['iss'], (str, ustr)):
+            raise TokenValidationError('Issuer (iss) claim must be a string present in the access token')
+        if payload['iss'] != self.iss:
+            raise TokenValidationError(
+                'Issuer (iss) claim mismatch in the access token; expected "{}", '
+                'found "{}"'.format(self.iss, payload['iss']))
+
+        # Subject
+        if 'sub' not in payload or not isinstance(payload['sub'], (str, ustr)):
```

Comment:
where is this requirement coming from? I couldn't find any article/doc mentioning the `sub` claim should be validated when checking an access token. Only found signature, time-related claims, audience and issuer.